0002-crypto-conditions/0002-crypto-conditions.md

```diff
@@ -114,312 +114,288 @@ This specification is a part of the [Interledger Protocol](https://interledger.o
 
-    cc:0:3:dB-8fb14MdO75Brp_Pvh4d7ganckilrRl13RS_UmrXA:66
+### Fulfillment Format {#preimage-sha-256-condition-type-fulfillment}
 
-Example fulfillment:
+The fulfillment payload is simply the preimage.
 
-    cf:0:VGhlIG9ubHkgYmFzaXMgZm9yIGdvb2QgU29jaWV0eSBpcyB1bmxpbWl0ZWQgY3JlZGl0LuKAlE9zY2FyIFdpbGRl
+### Example {#preimage-sha-256-example}
 
+TODO
 
 ## PREFIX-SHA-256 {#prefix-sha-256-condition-type}
-PREFIX-SHA-256 is assigned the type ID 1. It relies on the SHA-256 and PREFIX feature suites which corresponds to a feature bitmask of 0x05.
+PREFIX-SHA-256 is assigned the type ID 1. It relies on the availability of the SHA-256 digest algorithm and is a compound crypto-condition type.
 
-Prefix conditions provide a way to effective narrow the scope of other conditions. A condition can be used as the fingerprint of a public key to sign an arbitrary message. By creating a prefix subcondition we can narrow the scope from signing an arbitrary message to signing a message with a specific prefix.
+Prefix crypto-conditions provide a way to effectively narrow the scope of other crypto-conditions. 
 
-When a prefix fulfillment is validated against a message, it will prepend the prefix to the provided message and will use the result as the message to validate against the subfulfillment.
+A condition is the fingerprint of a public key used to sign an arbitrary message. By creating a prefix crypto-condition that wraps another condition we can narrow the scope from signing an arbitrary message to signing a message with a specific prefix.
 
-### Condition {#prefix-sha-256-condition-type-condition}
-The fingerprint of a PREFIX-SHA-256 condition is the SHA-256 digest of the fingerprint contents given below:
+We can also use the prefix condition in contexts where there is an empty message used for validation of the fulfillment so that we can reuse the same key pair for multiple fulfillments, each with a different prefix, and therefore generate a unique condition each time.
 
-    PrefixSha256FingerprintContents ::= SEQUENCE {
+Implementations MUST prepend the prefix to the provided message and will use the resulting value as the message to validate the sub-fulfillment.
+
+### Condition Format {#prefix-sha-256-condition-type-condition}
+
+The fingerprint of a PREFIX-SHA-256 condition is the SHA-256 digest of the fingerprint contents as defined below:
+
+    PrefixSha256ConditionFingerprintContents ::= SEQUENCE {
       prefix OCTET STRING,
-      condition Condition
+      subcondition Condition
     }
 
-
 prefix
 : is an arbitrary octet string which will be prepended to the message during validation.
 
-condition
-: is the subcondition which the subfulfillment must match.
+subcondition
+: is the sub-condition derived from the sub-fulfillment.
+
+
+### MaxFulfillmentLength {#prefix-sha-256-condition-type-maxfulfillmentlength}
 
+The maxFulfillmentLength is the size, in bytes, of the prefix plus the maxFulfillmentLength length of the sub-condition.
```

Comment:
Just pointing out that, strictly speaking, this field isn't necessary in this instance, since we have the information to derive this directly in the condition fingerprint: the prefix, and the subcondition. Of course, we can't just skip it, so...

Food for thought: do we gain anything by storing the _SHA256_ digest, à la `preimage-sha-256` instead?